spec/formatting.md

```diff
@@ -14,9 +14,11 @@ To resolve the value of a Variable,
 its Name is used to identify either a local variable,
 or a variable defined elsewhere.
 If a local variable and an externally defined one use the same name,
-the local variable takes precedence.
+the local variable takes precedence after its variable declaration.
+A variable declaration may refer in its expression
+to a previous or external variable with the same name as the declaration's target.
```

Comment:
I think this is good, but I think we could be more explicit. Perhaps instead of this paragraph we should say:

> Each `declaration` is processed sequentially. 
> A `declaration` **_may_** refer to previously declared `variables` (including itself) or to external variables in its `expression`. 
> A `declaration` that shares a name with a previous `declaration` or to an external variable overwrites the preceding or external variable.

---

Note the use of 2119 keywords and keyword quoting, which is our standard.

Reply:
I don't feel strongly about the wording, although I like that @aphillips's suggestions is slightly more explicit. That said:

> Each declaration is processed sequentially.

I agree that processing declarations should generally work sequentially. At the same time, do we need to be prescriptive about it in terms of the implementation?

> A declaration may refer to previously declared variables (including itself) or to external variables in its expression.

Here, OTOH, I think we should rather say `(excluding itself)`. Otherwise we're getting into a teritorry of circular references.

Reply:
Apologies for nitpicking, but I think it would be helpful for us to use valid syntax when arguing about spec details. To wit:

```
let $foo = {|some value|}
let $foo = {$foo :functionA}
let $foo = {$foo :functionB}
let $foo = {||}
```

Reply:
So, it's not really referring to *itself*, is it? It's referring to a different value, by identity, but with the same name. (I guess it really depends on whether we think of `let $foo` as a new variable, or a new name binding...)